### PR TITLE
AWS ML Blog edits

### PR DESCRIPTION
**Pull request by dougkelly**
* Reduce dataset chunking size for GPU OOM errors
* Typo fixes
* Add explanatory text from blog draft

SageMaker - @philschmid

### CODE REVIEW

File: sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb [hunk 1]
```diff
@@ -201,7 +196,7 @@
     "lm_dataset = dataset.map(\n",
     "    lambda sample: tokenizer(sample[\"text\"]), batched=True, remove_columns=list(dataset.features)\n",
     ").map(\n",
-    "    partial(chunk, chunk_length=2048),\n",
```
Comment (philschmid) [inline sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: can we try decreasing the batch size. It is better to train on the full sequence length

File: sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb [hunk 2]
```diff
@@ -428,24 +444,572 @@
...
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": []
   }
  ],
  "metadata": {
+  "availableInstances": [
```
Comment (philschmid) [inline sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:453]: Thats should not be there.

File: sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb (hunk 1 above)
Comment (philschmid) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: or pinning the `peft` version to `0.2.0`

Comment (dougkelly) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: I tried a few runs pinning to peft 0.2.0 and received the following error after ~5 hours of training:

ErrorMessage "│   1908 │   │   │   raise ValueError("training mode is expected to be boolean │
 │   1909 │   │   self.training = mode                                          │
 │   1910 │   │   for module in self.children():                                │
 │ ❱ 1911 │   │   │   module.train(mode)                                        │
 │   1912 │   │   return self                                                   │
 │   1913 │                                                                     │
 │   1914 │   def eval(self: T) -> T:                                           │
 │                                                                              │
 │ /opt/conda/lib/python3.9/site-packages/torch/nn/modules/module.py:1911 in    │
 │ train                                                                        │
 │ /opt/conda/lib/python3.9/site-packages/peft/tuners/lora.py:417 in train      │
 │   414 │   │   if not mode and self.merge_weights and not self.merged:        │
 │   415 │   │   │   # Merge the weights and mark it                            │
 │   416 │   │   │   if self.r > 0 and any(self.enable_lora):                   │
 │ ❱ 417 │   │   │   │   delta_w = F.conv1d(                                    │
 │   418 │   │   │   │   │   self.lora_A.weight.data.unsqueeze(0),              │
 │   419 │   │   │   │   │   self.lora_B.weight.data.unsqueeze(-1),             │
 │   420 │   │   │   │   │   groups=sum(self.enable_lora),                      │
 ╰──────────────────────────────────────────────────────────────────────────────╯
 RuntimeError: Expected 4-dimensional input for 4-dimensional weight [8192, 8, 1,
 1], but got 3-dimensional input of size [1, 16, 4096] instead"

Comment (dougkelly) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: 'per_device_train_batch_size': 1 so I don't believe lowering the batch size is an option.

File: sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb (hunk 2 above)
Comment (dougkelly) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:453]: Sorry, sloppy on my part. Will remove on next commit.

File: sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb (hunk 1 above)
Comment (dougkelly) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: Other items tried just to have the full debugging trail here:
- Pinning to v0.3.0 stable
- Setting os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:X" at top of run_clm.py script with various values 516, 1500

Continuing to get GPU memory error -
ErrorMessage "OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB (GPU 0; 22.19
GiB total capacity; 15.37 GiB already allocated; 1.62 GiB free; 19.37 GiB
reserved in total by PyTorch)

Reducing the dataset chunk_length was sufficient to address the GPU OOM issues. I hear you on training for the full sequence length, will continue to explore workarounds.

Comment (philschmid) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: I tested with peft 0.2.0 it works. Can you confirm? i would pin then.

Comment (dougkelly) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:204]: Thanks for testing out @philschmid. I think I was encountering the weird errors above due to testing artifacts from tinkering but I was able to confirm pinning to peft `v0.2.0` solved the OOM issue.

File: sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb (hunk 2 above)
Comment (dougkelly) [reply sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb:453]: Spent like 2 hours trying to remove all this extra metadata SM Studio adds, I ended up just opening a superseding PR #373 to address. Separately reaching out to SM Studio team for workaround / remove this going forward.

**Comment by dougkelly** [comment]
Closing this PR. Superseded by PR #373 which pins PEFT to v0.2.0 and reverses changes to `chunk_length` the extra metadata dump from SM Studio.

**Pull request closed without merging**